src/extension.ts

```diff
@@ -26,9 +31,51 @@ export async function activate (context: vscode.ExtensionContext) {
 	context.subscriptions.push(...subscriptions);
 }
 
-// TO-DO: add call to Hydrate Python script
-function hydrateCluster () {
+/**
+ * This method spawns a subprocess that calls Hydrate, which creates a Fabrikate 
+ * component.yaml file in the same directory as the Hydrate clone.
+ * 
+ * It runs Hydrate on the kubeconfig file associated with the clusters in the sidebar,
+ * and it currently uses the default values for the rest of the options.
+ */
+function hydrateCluster (target?: any) {
+	let kubeconfig = getKubeConfig();
+	let isErr = false;
+
+	console.log('Hydrating...');
+
+	const subprocess = spawn('python3', ['-m', 'hydrate.hydrate', '-k', kubeconfig, 'run'], {
+		cwd: HOME
+	});
+	
+	// outputs stdout of the Hydrate subprocess
+	subprocess.stdout.on('data', function (data: any) {
+		if (!isErr) {
+			console.log(data.toString());
+			console.log('Done!');
+		}
+	});
+
+	// outputs errors of subprocess
+	subprocess.stderr.on('data', function (data: any) {
+		let err = data.toString();
+		console.log(err);
+		isErr = true;
+	});
+
+}
+
+function getKubeConfig () {
```

Comment:
use a typescript return type to enforce type safety elsewhere - i think this would be a return type of `string` so your method signature would become:
```
function getKubeConfig (): string {
  // implementation
}
```